### PR TITLE
In DetailView, delete button and edit button are functioning

### PR DESCRIPTION
**Pull request by mohammra**
In DetailView, when someone press delete option then the saved note delete from coredata, and when someone press on edit otion, they can manually edit the text as they wish

### CODE REVIEW

File: Voice Note/Views/DetailView.swift [hunk 1]
```diff
@@ -52,26 +65,70 @@ struct DetailView: View {
...
+    
+    private func toggleEditing() {
+        if isEditing {
+            saveEditedText()
+        }
+        isEditing.toggle()
+    }
+    
+    private func saveEditedText() {
+        let context = PersistenceController.shared.container.viewContext
```
Comment (anwar-ulhaq) [inline Voice Note/Views/DetailView.swift:77]: No. ask CoreData service to provide context

File: Voice Note/Views/DetailView.swift [hunk 2]
```diff
@@ -52,26 +65,70 @@ struct DetailView: View {
...
+        voiceNote.text = editText
+        do {
+            try context.save()
+        } catch {
+            print("Error saving edited text: \(error)")
+        }
+    }
+    
+    private func deleteVoiceNote() {
+        let context = PersistenceController.shared.container.viewContext
```
Comment (anwar-ulhaq) [inline Voice Note/Views/DetailView.swift:87]: Call coredata to provide context

File: Voice Note/Views/ListItem.swift [hunk 3]
```diff
@@ -11,6 +11,13 @@ struct ListItem: View {
     @State private var isDeleteAlert = false
     @State private var defaultSelect = "None"
     @State private var showDetail = false
+
+    private func formattedDuration(for duration: Int) -> String {
```
Comment (anwar-ulhaq) [inline Voice Note/Views/ListItem.swift:15]: Check TimeDuration.swift

File: Voice Note/Views/RecordingListView.swift [hunk 4]
```diff
@@ -10,21 +10,19 @@ import CoreLocation
 
 struct RecordingListView: View {
     @State var toast:ToastView? = nil
-    //@Environment(\.managedObjectContext) private var managedObjectContext
     @Environment(\.coreData) private var coreDataService: CoreDataService
```
Comment (anwar-ulhaq) [inline Voice Note/Views/RecordingListView.swift:13]: `@Environment(\.managedObjectContext) private var managedObjectContext` was there if we need it at some point

File: Voice Note/Views/DetailView.swift (hunk 1 above)
Comment (anwar-ulhaq) [reply Voice Note/Views/DetailView.swift:77]: `let context = CoreDataService. getManageObjectContext()`
To get context from View, use 
`@Environment(\.managedObjectContext) private var managedObjectContext`
`@Environment(\.coreData) private var coreDataService: CoreDataService`

File: Voice Note/Views/DetailView.swift [hunk 5]
```diff
@@ -4,43 +4,56 @@
...
+                    .padding()
+                    .frame(maxWidth: .infinity)
+                    .background(Color(textContainer))
+                    .cornerRadius(20)
+                    .padding()
+            }
             
             RecordingCardView().padding(15)
-            Text("Duration: 23s")
+            Text("Duration: \(voiceNote.duration)s")
```
Comment (anwar-ulhaq) [inline Voice Note/Views/DetailView.swift:42]: I think, I need to make some changes to duration in CoreData. Lets see later.

**Review by giaongo — APPROVED** [review]
Changes approved :)

**Pull request closed without merging**